iree/turbine/kernel/compiler/dispatch_codegen.py

```diff
@@ -181,9 +181,16 @@ def abi_type(binding: BindingDesc):
 
             # Define the export.
             with InsertionPoint.at_block_begin(self._exe_block):
+                index_type = IndexType.get()
                 export_op = stream_d.ExecutableExportOp(name, name)
                 export_block = export_op.workgroup_count.blocks.append(
-                    *([b.as_mlir_type() for b in dynamic_dim_bindings])
+                    *(
+                        [
+                            index_type
+                            for b in dynamic_dim_bindings
+                            + [b for b in scalar_bindings if b.symbol_type is not None]
```

Comment:
why do you need
```
if b.symbol_type is not None
```
?

Reply:
`b.symbol_type is None` means it's just a normal scalar arg without any attached symbol and they cannot participate in grid calculations.

Reply:
Could you create a function with your explanation above:
```
def _is_scalar_and_not_symbol_bound(b: ...):
       """
        ..... b.symbol_type is None means it's just a normal scalar arg without any attached symbol and they cannot participate in grid calculations ....
       """
       return b.symbol_type is not None
```

which would make it more clear?

Reply:
done